### PR TITLE
Fix FIST updates

### LINKED ISSUE #65: Corner cases in FIST algorithm
The FIST implementation we have is reasonably robust, but it is still failing with some difficult polygonal areas. In our test suite we have a bunch of polygonal area examples in the `data` directory that are loaded for testing. Tests are passing for most polygonal areas in the family `poly1...5.line`, for example:

# poly1.line
image
image

# poly2.line
image
image

# poly3.line
But some cases like `poly3.line` are not passing the visual test:

image
image

Notice how some edges cross. That shouldn't be the case.

With the latest changes, we can also handle polygonal areas with holes in the `holes1...5.line` family, for example:

# hole2.line
image
image

# hole1.line
But some cases like `hole1.line` have crossing edges too:

image
image

Notice how the left inner hole is crossed by edges.

Finally we have polygonal areas in the `smooth1...5.line` family that are working fine:

# smooth2.line
image
image

# smooth1.line
But similarly, some cases fail with crossing edges:

image
image

It would be nice to dive in debug mode to figure out what is happening. Both `PolyArea` and `UnstructuredMesh` can be plotted directly with our current plot recipes so that helps too in the debugging process.

If someone can read the FIST paper by Held and then read our implementation, which basically follows the exact same notation, that would be great. I think I've implemented all the ingredients except for one or two corner cases they discuss, which may be related to the issues we are seeing.

If someone wants to dive in, I can help explaining the current implementation in a video chat for example.

### PR DESCRIPTION
**Pull request by juliohm**
Fix #65 Supersede #66

### CODE REVIEW

**Comment by juliohm** [comment]
@serenity4 I've started this clean PR with a simplified version of the fix you proposed in #66 . I messed up the other PR trying to improve it. Also, we can work on the missing features before we add tests for this one. For example, the missing `vertices(polyarea)` method.

Right now the plan is to make sure that the results look correct visually. There is a case `smooth2.line` that is hanging in single precision, and I realized that some tests in the test/polytopes.jl are still failing in single precision.

Can we merge this simple fix here, and continue working on separate PRs before we come back to handle `smooth2.line`?

**Review by serenity4 — APPROVED** [review]
Looks good!

File: src/discretization/fist.jl [hunk 1]
```diff
@@ -50,8 +50,13 @@ function discretize(polyarea::PolyArea, ::FIST)
...
       # 3. update 𝒬 near clipped ear
-      isear(𝒫, i)   && (𝒬 = 𝒬 ∪ [mod1(i,n)])
-      isear(𝒫, i+1) && (𝒬 = 𝒬 ∪ [mod1(i+1,n)])
+      for j in (i-1, i)
+        if isear(𝒫, j)
+          𝒬 = 𝒬 ∪ [mod1(j,n)]
+        else
+          𝒬 = setdiff(𝒬, [mod1(j,n)])
+        end
+      end
```
Comment (serenity4) [inline src/discretization/fist.jl]: Quick note: this is not in-place, so likely slower that the original PR with unnecessary allocations. But we can do an optimization pass later and not worry about that now.

File: src/discretization/fist.jl [hunk 2]
```diff
@@ -50,8 +50,13 @@ function discretize(polyarea::PolyArea, ::FIST)
       𝒫 = Chain(points[inds])
       n = nvertices(𝒫)
       # 3. update 𝒬 near clipped ear
-      isear(𝒫, i)   && (𝒬 = 𝒬 ∪ [mod1(i,n)])
-      isear(𝒫, i+1) && (𝒬 = 𝒬 ∪ [mod1(i+1,n)])
+      for j in (i-1, i)
+        if isear(𝒫, j)
+          𝒬 = 𝒬 ∪ [mod1(j,n)]
+        else
+          𝒬 = setdiff(𝒬, [mod1(j,n)])
```
Comment (serenity4) [inline src/discretization/fist.jl]:
```suggestion
          setdiff!(𝒬, [mod1(j,n)])
```

File: src/discretization/fist.jl (hunk 1 above)
Comment (juliohm) [reply src/discretization/fist.jl]: Yep. I think we can work on performance optimizations in a separate pass. We are still in the process of making sure the algorithm is working smoothly with various example polygons.
